msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/PrivateMessageManagerImpl.java

```diff
@@ -1574,6 +1585,40 @@ public void markMessageAsReadForUser(final PrivateMessage message, final String
     }
   }
 
+  public void sendConfirmationEmail(PrivateMessage pvtMessage, PrivateMessageRecipientImpl pvtRecipient){
+    String defaultEmail = serverConfigurationService.getString("setup.request","postmaster@" + serverConfigurationService.getServerName());
+    String currentUserAsString = currentUserAsString(pvtMessage, false);
+    String contextId = getContextId();
+
+    Message savedMessage = saveMessage(pvtMessage, false, contextId, currentUserAsString);
+
+    String systemEmail = "";
+    try {
+      User currentUser = currentUser(pvtMessage, false);
+      List<InternetAddress> replyEmail  = new ArrayList<>();
+      systemEmail = getSystemAndReplyEmail(defaultEmail, currentUser, savedMessage, replyEmail, contextId);
+    } catch (MessagingException e) {
+      log.warn("PrivateMessageManagerImpl.sendConfirmationEmail: exception: " + e.getMessage(), e);
+    }
+
+    List additionalHeaders = new ArrayList(1);
+    additionalHeaders.add("Content-Type: text/html; charset=utf-8");
+    additionalHeaders.add("From: " + systemEmail);
+    additionalHeaders.add("Subject: " + rb.getFormattedMessage("pvt_read_receipt_email_subject", pvtMessage.getTitle()));
+
+    User user = null;
+    User messageCreator = null;
+    try {
+      user = userDirectoryService.getUser(pvtRecipient.getUserId());
+      messageCreator = userDirectoryService.getUser(pvtMessage.getCreatedBy());
+    } catch (UserNotDefinedException e) { }
+
+    DateFormat df = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss");
```

Comment:
This is hardcoded, do not hardcode date formats.

Reply:
Look at this
https://github.com/sakaiproject/sakai/blob/fe6ac1e2b32f57e4503efdbaf2ee467bb8b8a184/msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionTopicBean.java#L1195